packages/@n8n/nodes-langchain/nodes/agents/Agent/V2/AgentV2.node.ts

```diff
@@ -114,7 +114,7 @@ export class AgentV2 implements INodeType {
 				((hasOutputParser, needsFallback) => {
 					${getInputs.toString()};
 					return getInputs(hasOutputParser, needsFallback)
-				})($parameter.hasOutputParser === undefined || $parameter.hasOutputParser === true, $parameter.needsFallback === undefined || $parameter.needsFallback === true)
+				})($parameter.hasOutputParser === undefined || $parameter.hasOutputParser === true, $parameter.needsFallback !== undefined && $parameter.needsFallback === true)
```

Comment:
Thanks for fixing. Approved to get this in quickly and hotfix it. 
But would be great to add tests that would catch this if possible